### DISCUSSION TITLE
Access to the model object or request

### DISCUSSION THREAD

**Discussion by pasindujr** [rappasoft/laravel-livewire-tables#1408] [Q&A] [answered]
Hi. I use a data table inside a view which is rendered when visiting an URL like this: http://127.0.0.1:8000/events/106. I want to get the data of single "event". I passed the data from controller to view and passed it again to the livewire datatable view. I'm struggling to get passed data inside datatable class or any other way to write a query to get the needed data, like accessing the "106" in this case. TIA!

**Comment by lrljoe** [comment]
You can use the builder() approach within your table component.

Please share the content of your table component and I'll try to advise

**Reply by pasindujr** [reply]
Thanks for the response. Below is the code for the table. I know about `builder()` method, but how to access the request to get the specific id?
```php
<?php

namespace App\Livewire;

use Illuminate\Database\Eloquent\Builder;
use Rappasoft\LaravelLivewireTables\DataTableComponent;
use Rappasoft\LaravelLivewireTables\Views\Column;
use App\Models\SeatAllocation;
use Rappasoft\LaravelLivewireTables\Views\Columns\BooleanColumn;

class SeatAllocationTable extends DataTableComponent
{
    public function builder(): Builder
    {
        return SeatAllocation::query();
    }

    public function configure(): void
    {
        $this->setPrimaryKey('id');
    }

    public function columns(): array
    {
        return [
            Column::make("Id", "id")
                ->sortable(),
            Column::make("Event id", "event_id")
                ->sortable(),
            Column::make("Seat number", "seat_number")
                ->sortable(),
            BooleanColumn::make("Is reserved", "is_reserved")
                ->sortable(),
        ];
    }
}

```

**Reply by lrljoe** [reply, marked as answer]
What's your plan here, do you want to pass it to a view, use it to do something else, or what?

Is it as simple as, a parent controller/component is receiving the https://127.0.0.1:8000/events/106 request.

You are then passing the 106 (eventID?) to the table and you want to restrict the query to those relating to the eventID?

**Reply by pasindujr** [reply]
Thank you very much! Now it's working properly. So this table component is just like a regular Livewire component, isn't it?

**Reply by lrljoe** [reply]
Pretty much!  There's a lot that goes on behind the scenes, but it is just a Livewire Component with several dozen traits ultimately.
